tests/lib/nrf_modem_lib/lte_connectivity/src/lte_connectivity_test.c

```diff
@@ -365,80 +435,221 @@ void test_options_get_should_return_error_if_wrong_option_name_is_used(void)
 
+	/* Activate PDN */
 	pdn_event_handler_callback(0, PDN_EVENT_ACTIVATED, 0);
 
+	/* Verify IPv4 was not added */
 	TEST_ASSERT_FALSE(net_if_ipv4_get_global_addr(net_if, NET_ADDR_PREFERRED));
 }
 
-void test_pdn_ipv4_address_should_be_removed_on_deactivated(void)
-{
-	ipv4_address_add();
+/* IPv6 */
 
+/* Verify that PDN activation and deativation adds and removes an IPv6 address if provided */
+void test_pdn_triggers_ipv6_address_changes(void)
+{
 	struct net_if *net_if = net_if_get_default();
 
-	__cmock_nrf_modem_is_initialized_ExpectAndReturn(1);
+	/* Expect lte_connectivity to check for an IPv6 address when PDN is activated */
+	__mock_nrf_modem_at_scanf_ExpectAndReturn(
+			"AT+CGPADDR=0", "+CGPADDR: %*d,\"%46[.:0-9A-F]\",\"%46[:0-9A-F]\"", 2);
 
-	pdn_event_handler_callback(0, PDN_EVENT_DEACTIVATED, 0);
+	/* Provide one */
+	__mock_nrf_modem_at_scanf_ReturnVarg_string("");
+	__mock_nrf_modem_at_scanf_ReturnVarg_string("2001:8c0:5140:801:1b4d:4ce8:6eed:7b69");
 
-	TEST_ASSERT_TRUE(net_if_flag_is_set(net_if, NET_IF_DORMANT));
-	TEST_ASSERT_FALSE(net_if_ipv4_get_global_addr(net_if, NET_ADDR_PREFERRED));
-}
+	/* Activate PDN */
+	pdn_event_handler_callback(0, PDN_EVENT_IPV6_UP, 0);
 
-/* IPv6 */
+	/* Verify IPv6 was added */
+	TEST_ASSERT_TRUE(net_if_ipv6_get_global_addr(NET_ADDR_PREFERRED, &net_if));
 
-void test_pdn_ipv6_address_should_be_added_on_ipv6_up(void)
-{
-	ipv6_address_add();
+	/* Let events fire */
+	k_sleep(K_MSEC(10));
+
+	/* Dectivate PDN */
+	pdn_event_handler_callback(0, PDN_EVENT_DEACTIVATED, 0);
+
+	/* Verify that IPv6 was removed */
+	TEST_ASSERT_FALSE(net_if_ipv6_get_global_addr(NET_ADDR_PREFERRED, &net_if));
 }
 
-void test_pdn_ipv6_fails_to_be_added_on_ipv6_up(void)
+
+/* Verify that PDN activation does not add an IPv6 address if none is provided */
+void test_pdn_does_not_add_nonexistant_ipv6(void)
 {
 	struct net_if *net_if = net_if_get_default();
 
+	/* Expect lte_connectivity to check for an IPv4 address when PDN is activated */
 	__mock_nrf_modem_at_scanf_ExpectAndReturn(
 			"AT+CGPADDR=0", "+CGPADDR: %*d,\"%46[.:0-9A-F]\",\"%46[:0-9A-F]\"", 0);
+
+	/* Do not provide one */
 	__cmock_lte_lc_func_mode_set_ExpectAndReturn(LTE_LC_FUNC_MODE_DEACTIVATE_LTE, 0);
 
+	/* Activate PDN */
 	pdn_event_handler_callback(0, PDN_EVENT_ACTIVATED, 0);
 
+	/* Verify IPv6 was not added */
 	TEST_ASSERT_FALSE(net_if_ipv6_get_global_addr(NET_ADDR_PREFERRED, &net_if));
 }
 
-void test_pdn_ipv6_address_should_be_removed_on_ipv6_down(void)
+/* CEREG and PDN readiness */
+
+/* Verify that the iface is inactive before PDN and CEREG are ready, becomes active
+ * afterwards, and becomes inactive again if either is lost.
+ *
+ * Verify also that LTE_LC_NW_REG_SEARCHING is ignored
+ *
+ * k_sleep calls are not strictly necessary since iface state changes are instantaneous,
+ * but we perform them so that the NET_MGMT event queue is not overflowed during the test.
+ */
+void test_iface_pdn_and_cereg_tracking(void)
```

Comment:
Sure, I can split it up. I thought about doing so for some of these tests, but I thought the added code bloat would not be worth the marginal gains of test specificity -- especially since failed tests indicate what line they failed on.

This is your suite, though, so I'll defer to your preferences